common/test/acceptance/tests/lms/test_teams.py

```diff
@@ -980,6 +984,7 @@ def test_join_team(self):
         """
         self._set_team_configuration_and_membership(create_membership=False)
         self.team_page.visit()
+        self.team_page.wait_for_element_presence('.join-team .action-primary', 'Waiting on join button', 30)
         self.assertTrue(self.team_page.join_team_button_present)
```

Comment:
Another approach here is to wait for self.team_page.join_team_button_present to return True. I'd probably also make that a helper method in the team_page class itself (that code already exists in one place there, so you could refactor to create a helper method).

Something like wait_for_join_team_button_presence.

Then you don't need the assert that the button is present afterwards because the wait won't return until the link is present.